### PR TITLE
Improve `CONTRIBUTING.md` instructions

### PR DESCRIPTION
**Pull request by Reminiscent**
This PR improves the `CONTRIBUTING.md` by:
- Adding missing steps for initializing submodules with `git submodule update`.
- Providing clear instructions for connecting to PostgreSQL using `psql`.
- Explaining how to enable the `pg_mooncake` extension after connecting to `psql`.
- Adding the test process

### CODE REVIEW

**Comment by Reminiscent** [comment]
@dpxcc I find some informations are missing when I try to develop the extension. Add a PR to improve it. PTAL

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -17,10 +17,18 @@ This will create an isolated Workspace in vscode, including all tools required t
...
 ```bash
+git submodule update --init --recursive
 make debug
 make install
 ```
-then connect to Postgres using `psql`.
+Then, connect to PostgreSQL using psql:
+```bash
+psql -U postgres -d mooncake
+```
```
Comment (dpxcc) [inline CONTRIBUTING.md:27]: PostgresSQL -> Postgres for consistency
I think `psql` should just work if you are using dev container, no need for `-U postgres -d mooncake`

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -33,5 +41,8 @@ This pid (1219 in this case) indicates the process that you should attach the de
 
 3. Set Breakpoints and Debug: With the debugger attached, you can set breakpoints within the code. This allows you to step through the code execution, inspect variables, and fully debug the Postgres instance running in your container.
 
+## Testing
+* Tests use standard regression tests for Postgres extensions. To run tests, run make installcheck.
```
Comment (dpxcc) [inline CONTRIBUTING.md:45]: make installcheck -> `make installcheck`

**Comment by Reminiscent** [comment]
@dpxcc Thanks for you review. Updated! PTAL

**Pull request merged**